crates/astria-proto/src/lib.rs

```diff
@@ -1,12 +1,15 @@
+pub use prost::{
+    DecodeError,
+    EncodeError,
+    Message,
+};
+
 #[allow(warnings)]
 #[allow(unreachable_pub)]
 #[rustfmt::skip]
 mod proto;
 
-pub use proto::generated::{
-    execution,
-    primitive,
-    sequencer,
-};
+#[cfg(feature = "native")]
+pub mod native;
```

Comment:
I am torn on nesting this here. I wonder if we should introduce a `astria-rpc` or `astria-api` crate and make the protos a submodule of it.

I believe there are good reasons to have the native types and their protobuf counterparts live in the same crate (mainly to avoid issues around orphan rules), but having the "native" types underneath something called `-proto` seems confusing?

Reply:
agreed, we should probably rename this crate and put `proto` as a module inside it

Reply:
I like `astria-core` better, I have a dislike for crates and modules named `*-types`.

I would however leave the renaming to a followup-pr to not blow up this PR (and to not mix renaming + new business logic).

And then we can do better bikeshedding on names. 😂

What do you think?